src/uiScreens.c

```diff
@@ -172,7 +173,34 @@ void ui_displayStakingInfoScreen(
 	);
 }
 
-void ui_displayAmountScreen(
+void ui_displayTokenNameScreen(
+        token_amount_t* token,
+        ui_callback_fn_t callback
+)
+{
+	if (str_isAsciiPrintableBuffer(token->assetName, token->assetNameSize)) {
+		char name[ASSET_NAME_SIZE_MAX + 1];
+		ASSERT(token->assetNameSize + 1 <= SIZEOF(name));
+		os_memmove(name, token->assetName, token->assetNameSize);
+		name[token->assetNameSize] = '\0';
+
+		bool isEmpty = (token->assetNameSize == 0);
+
+		ui_displayPaginatedText(
+		        (isEmpty) ? "Token name (empty):" : "Token name (ASCII):",
```

Comment:
```suggestion
		        (isEmpty) ? "Token name is empty" : "Token name (ASCII):",
```

Reply:
also, in Trezor we use the term "asset name" which also aligns with the terminology used in developer docs: https://developers.cardano.org/en/development-environments/native-tokens/multi-asset-tokens-explainer/ so I'd prefer showing "asset name" over "token name"